elasticdl/python/tests/test_helper.py

```diff
@@ -0,0 +1,176 @@
+                example_dict = {
+                    "image": tf.train.Feature(
+                        bytes_list=tf.train.BytesList(value=[image])
+                    ),
+                    "label": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=[label])
+                    ),
+                }
+            elif dataset == "frappe":
+                feature = np.random.randint(5383, size=(shape,))
+                label = np.random.randint(2, size=(1,))
+                example_dict = {
+                    "feature": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=feature)
+                    ),
+                    "label": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=[label])
+                    ),
+                }
+            elif dataset == "test_module":
+                x = np.random.rand(shape).astype(np.float32)
+                y = 2 * x + 1
+                example_dict = {
+                    "x": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=x)
+                    ),
+                    "y": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=y)
+                    ),
+                }
+            elif dataset == "image_default":
+                image = np.random.rand(np.prod(shape)).astype(np.float32)
+                label = np.ndarray([1], dtype=np.int64)
+                label[0] = np.random.randint(0, 10)
+                example_dict = {
+                    "image": tf.train.Feature(
+                        float_list=tf.train.FloatList(value=image)
+                    ),
+                    "label": tf.train.Feature(
+                        int64_list=tf.train.Int64List(value=[label])
+                    ),
+                }
+            else:
+                raise ValueError("Unknown dataset name %s." % dataset)
+
+            example = tf.train.Example(
+                features=tf.train.Features(feature=example_dict)
+            )
+            f.write(example.SerializeToString())
+    return temp_file.name
+
+
+def distributed_train_and_evaluate(
+    feature_shape,
+    model_zoo_path,
+    model_def,
+    model_params="",
+    training=True,
+    dataset="image_default",
+    callback_classes=[],
+    use_async=False,
+    get_model_steps=1,
+):
+    """
+    Run distributed training and evaluation with a local master.
+    grpc calls are mocked by local master call.
+    """
+    job_type = (
+        JobType.TRAINING_WITH_EVALUATION
+        if training
+        else JobType.EVALUATION_ONLY
+    )
+    batch_size = 8 if dataset == "imagenet" else 16
+    worker = Worker(
+        1,
+        job_type,
+        batch_size,
+        model_zoo_path,
+        model_def=model_def,
+        model_params=model_params,
+        channel=None,
+        get_model_steps=get_model_steps,
+    )
+
+    if dataset in ["imagenet", "frappe"]:
```

Comment:
Move these strings to some constants in `test_helper.py` (also appear in `create_recordio_file`) so other tests can reuse to avoid typos

Reply:
Done.